src/Microsoft.ML.AutoML/Experiment/Experiment.cs

```diff
@@ -196,6 +196,22 @@ public IList<TRunDetail> Execute()
                         "was reached, and the running MLContext was stopped. Details: {0}", e.Message);
                     return iterationResults;
                 }
+                catch (AggregateException e)
+                {
+                    // This exception is thrown when the IHost/MLContext of the trainer is canceled due to
+                    // reaching maximum experiment time. Simply catch this exception and return finished
+                    // iteration results. For some trainers, like FastTree, because training is done in parallel
+                    // in can throw multiple OperationCancelledExceptions. This causes them to be returned as an
+                    // AggregateException and misses the first catch block. This is to handle that case.
+                    if (e.InnerExceptions.All( exception => exception is OperationCanceledException))
```

Comment:
>e.InnerExceptions.All( exception => exception is OperationCanceledException [](start = 24, length = 75)

Is it possible to get a mix of exception in the list? I mean some be OperationCanceledException and some be other exceptions? if so, should change `All` to `Any`?

Reply:
nit: also spacing
```suggestion
                    if (e.InnerExceptions.Any(exception => exception is OperationCanceledException))
```

Reply:
I am not sure. Originally we weren't catching any errors there, then that was changed to only catch the cancelled exception. I am trying to keep the same behavior, just accounting for the cases when multiple cancelled exception are thrown so its an aggregate exception. If you think Any would be better I am happy to change it, that is just a behavior change and I'm not sure if we want that or not.

@JakeRadMSFT thoughts? @justinormont thoughts?

Reply:
I think it should be all because that keeps the current behavior of only catching the OperationCancelled behavior. I'm happy to change it if we think it should be Any though.